src/Giveaway.js

```diff
@@ -900,18 +900,24 @@ class Giveaway extends EventEmitter {
                 }
                 resolve(winners);
             } else {
-                const embed = this.fillInEmbed(options.messages.error.embed);
-                channel.send({
-                    content: this.fillInString(options.messages.error.content || options.messages.error),
-                    embeds: embed ? [embed] : null,
-                    allowedMentions: this.allowedMentions,
-                    reply: {
-                        messageReference:
-                            typeof options.messages.error.replyToGiveaway === 'boolean' ? this.messageId : undefined,
-                        failIfNotExists: false
-                    }
-                });
-                resolve([]);
+                if (options.rejectOnNoWinner) {
+                    reject(new Error('No valid participations, no new winner(s) can be chosen!'));
```

Comment:
here i've used a `Error` instance but I just noticed In all other places it's `reject(reason)`. I prefer using `Error` if u want me to change in all other places to `Error` or revert this one to only reason then lmk.